android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -752,4 +840,310 @@ public static ErcToken createEthereumErcToken() {
         }
         return tokens;
     }
+
+    public static String getAccountName(AccountInfo[] accounts, String address) {
+        String addressLowerCase = address.toLowerCase(Locale.getDefault());
+        for (AccountInfo account : accounts) {
+            if (account.address.toLowerCase(Locale.getDefault()).equals(addressLowerCase)) {
+                return account.name;
+            }
+        }
+
+        return stripAccountAddress(address);
+    }
+
+    public static void openTransaction(
+            TransactionInfo txInfo, EthJsonRpcController ethJsonRpcController, Activity activity) {
+        if (txInfo == null || txInfo.txHash == null || txInfo.txHash.isEmpty()) {
+            return;
+        }
+        assert ethJsonRpcController != null;
+        ethJsonRpcController.getChainId(chainId -> {
+            ethJsonRpcController.getAllNetworks(networks -> {
+                for (EthereumChain network : networks) {
+                    if (!chainId.equals(network.chainId)) {
+                        continue;
+                    }
+                    String blockExplorerUrl = Arrays.toString(network.blockExplorerUrls);
+                    if (blockExplorerUrl.length() > 2) {
+                        blockExplorerUrl =
+                                blockExplorerUrl.substring(1, blockExplorerUrl.length() - 1)
+                                + "/tx/" + txInfo.txHash;
+
+                        TabUtils.openUrlInNewTab(false, blockExplorerUrl);
+                        TabUtils.bringChromeTabbedActivityToTheTop(activity);
+                        break;
+                    }
+                }
+            });
+        });
+    }
+
+    public static void setUpTransactionList(AccountInfo[] accountInfos,
+            AssetRatioController assetRatioController, EthTxController ethTxController,
+            ErcTokenRegistry ercTokenRegistry, String assetSymbol, String contractAddress,
+            int assetDecimals, RecyclerView rvTransactions, OnWalletListItemClick callback,
+            Context context, String chainId) {
+        assert assetRatioController != null;
```

Comment:
maybe also put similar assertions for `ethTxController` and `ercTokenRegistry`  for consistency?

Reply:
there are asserts for them before their actual usage. I prefer to call it before usage as they could still be passed as null to the functions